packages/react/dialog/src/Dialog.tsx

```diff
@@ -247,7 +247,8 @@ const DialogContentModal = React.forwardRef((props, forwardedRef) => {
 
 const DialogContentNonModal = React.forwardRef((props, forwardedRef) => {
   const context = useDialogContext(CONTENT_NAME);
-  const isEscapeKeyDownRef = React.useRef(false);
+  const isPointerDownOutsideRef = React.useRef(false);
```

Comment:
This one is never getting reset in `DialogContentNonModal` or `PopoverContentNonModal`?

Reply:
They are reset during teardown / remount. `DropdownMenuRootContent` is unusual because this is handled one level lower by `Menu`, which means the top level state persists unless we reset manually.

Reply:
Teardown/remount isn't enough though, because you can have a pointer down but no close (so no teardown) for example when preventing default in `onPointerDownOutside` or in `onInteractOutside`.

Reply:
Good point and made me question another assumption, given the the case you describe we should probably be resetting when interacting _inside_ as well.

Reply:
It's really tricky, because basically what you want is reset it if there was a pointer down that didn't result in a close. But you don't really know this yet.

Reply:
Yeah.. the simplest way I could see was to explicitly handle ref clearing https://github.com/radix-ui/primitives/pull/799/commits/b74b6cfed252e009e5e9f952bd50cd2bb8519027 it's a bit verbose though.

Reply:
Yeah I don't love it haha, gonna have to think some more about this.
Any ideas @jjenzz?

Reply:
I'm not sure I'm fully understanding the issue. How do I replicate from within the `modality` branch?

- Open dialog
- Press close button
- Trigger doesn't auto focus?

I've just tried these steps in the `modality` branch and the trigger focuses fine so I'm missing something here for sure.

Reply:
It should focus if you close via the button. Try opening non-modal with keyboard and closing with return / space, it should focus the trigger but it doesn't

Reply:
I'm probably missing something but would this suffice?

![image](https://user-images.githubusercontent.com/175330/127167491-bbbc341a-ea2f-4d6f-809b-7f863b8ce379.png)


- Preventing default in `onPointerDownOutside` or `onFocusOutside` would prevent dismissing and `onInteractOutside` so would never set the ref to `true`.
- Dismissing via outside interaction only would set it to `true` which would close it and bring it back into it's default `false` state upon remount.

Reply:
> … and `onInteractOutside` so would never set the ref to true.

That's not the case right? `onInteractOutside` will run even if you call `preventDefault` in `onPointerDownOutside` or `onFocusOutside`.

Reply:
But now it's making me think, perhaps it's as simple as resetting it in `onInteractOutside` when `!event.defaultPrevented`?

Reply:
> `onInteractOutside` will run even if you call `preventDefault` in `onPointerDownOutside` or `onFocusOutside`.

Nope, which is really baffling to me but it doesn't. Because yah, I originally did what you suggested and console.logged `event.defaultPrevented` because I wanted to check that, but it never logged ...

Reply:
hmmm

![image](https://user-images.githubusercontent.com/1539897/127173569-111e21e3-a987-4c0b-84e5-b5dede3f656a.png)

The only thing behind the `defaultPrevented` check is `onDismiss`…

Reply:
If we want to be sure though, seems this should be all we need:

```diff
- else  isInteractOutsideRef.current = true;
+ else if (!event.defaultPrevented) isInteractOutsideRef.current = true;
```

Reply:
Yep, I have no idea what's going on there. It's made me worry that we might have a load of logic elsewhere that isn't running 😅

Reply:
If I change it to this then it calls it fine, but with the `event` param it doesn't

![image](https://user-images.githubusercontent.com/175330/127174562-315efdf3-dd27-4cd1-bb95-0127bbc9e380.png)

Reply:
Looks like there is a bug somewhere which we can fix separately, and we should definitely do the `!event.defaultPrevented` check in the meantime because `onInteractOutside` should be called.

Reply:
Did I open another can of worms here? haha. I've updated from the pro tips, is this what you had in mind? https://github.com/radix-ui/primitives/pull/799/commits/dfe047989d4ac612b3d1836fb9853d60415620b8

Reply:
haha I "think" that's correct, but man this is a real slippery area 😀 
I'm gonna have to take a deeper look at it and actually play with it myself a bit to get my head around it properly.